### ISSUE TITLE
Hide KeyedHashDB behind #[cfg(test)]

### ISSUE THREAD

**Issue by ordian** [paritytech/parity-ethereum#10387]
Currently, [`KeyedHashDB`](https://github.com/paritytech/parity-ethereum/blob/6bb106a784678cc2cadabfd621981371f477c48d/util/journaldb/src/traits.rs#L31) trait is used only in tests. To make it test-only, it may be a bit more complicated than just putting `#[cfg(test)]` on top of it, see https://github.com/paritytech/parity-ethereum/pull/10314#discussion_r258599139.

**Comment by ordian** [comment]
No longer relevant after #10929.
